src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -29,19 +28,23 @@ public void setPartition(final String partition) {
 
     @Nullable
     public String getPartition() {
-        return this.partition;
+        return partition;
     }
 
     public BatchItemResponse getResponse() {
-        return this.response;
+        return response;
     }
 
     public void setStep(final EventPublishingStep step) {
-        this.response.setStep(step);
+        response.setStep(step);
+    }
+
+    public EventPublishingStep getStep() {
+        return response.getStep();
     }
 
-    public synchronized void updateStatusAndDetail(final EventPublishingStatus publishingStatus, final String detail) {
-        this.response.setPublishingStatus(publishingStatus);
-        this.response.setDetail(detail);
+    public void updateStatusAndDetail(final EventPublishingStatus publishingStatus, final String detail) {
```

Comment:
There is no need for this combined update status and details anymore. It used to be like that due to a race condition protected by synchronized.